src/resources/promoSheet/promoSheetCollectionQuery.ts

```diff
@@ -0,0 +1,104 @@
+import { useQuery } from '@tanstack/react-query';
+
+import {
+  createQueryKey,
+  queryClient,
+  QueryConfig,
+  QueryFunctionArgs,
+  QueryFunctionResult,
+} from '@/react-query';
+
+import { arcDevClient } from '@/graphql';
+
+// Set a default stale time of 10 seconds so we don't over-fetch
+// (query will serve cached data & invalidate after 10s).
+const defaultStaleTime = 10_000;
```

Comment:
we can pump this waaaay up imo, if were not gonna show a sheet in < 5 min this should be around the same